src/providers/ogr/qgsgeopackagedataitems.cpp

```diff
@@ -771,8 +771,10 @@ QList<QString> QgsGeoPackageAbstractLayerItem::tableNames()
                );
       sqlite3_free( sql );
       if ( status != SQLITE_OK )
+      {  
         QgsDebugMsg( QStringLiteral( "There was an error reading tables from GPKG layer %1: %2" ).arg( mUri, QString::fromUtf8( errmsg ) ) );
       sqlite3_free( errmsg );
```

Comment:
??? are you added "sqlite3_free( errmsg )" in the if statement?

Reply:
seems you changed the code logic, try enclose with branches only QgsDebugMsg command:
      if ( status != SQLITE_OK )
      {  
        QgsDebugMsg( QStringLiteral( "There was an error reading tables from GPKG layer %1: %2" ).arg( mUri, QString::fromUtf8( errmsg ) ) );
      } 
      sqlite3_free( errmsg );